public/app/features/plugins/admin/hooks/usePlugins.tsx

```diff
@@ -58,22 +58,48 @@ type FilteredPluginsState = {
   plugins: CatalogPlugin[];
 };
 
-export const usePluginsByFilter = (searchBy: string, filterBy: string): FilteredPluginsState => {
-  const { loading, error, plugins } = usePlugins();
+type PluginsByFilterType = {
+  searchBy: string;
+  filterBy: string;
+  filterByType: string;
+};
+
+const filters = {
+  filterBy: (plugin: CatalogPlugin, filterBy?: string) =>
+    filterBy === 'installed' ? plugin.isInstalled : !plugin.isCore,
+
+  filterByType: (plugin: CatalogPlugin, filterByType?: string) =>
+    filterByType === 'all' || plugin.type === filterByType,
+
+  searchBy: (plugin: CatalogPlugin, searchBy?: string) => {
+    if (!searchBy) {
+      return true;
+    }
+    const fields: String[] = [];
+    if (plugin.name) {
+      fields.push(plugin.name.toLowerCase());
+    }
+
+    if (plugin.orgName) {
+      fields.push(plugin.orgName.toLowerCase());
+    }
 
-  const installed = useMemo(() => plugins.filter((plugin) => plugin.isInstalled), [plugins]);
+    return fields.some((f) => f.includes(searchBy.toLowerCase()));
+  },
+};
+
+export const usePluginsByFilter = (queries: PluginsByFilterType): FilteredPluginsState => {
+  const { loading, error, plugins } = usePlugins();
 
-  if (filterBy === 'installed') {
-    return {
-      isLoading: loading,
-      error,
-      plugins: applySearchFilter(searchBy, installed),
-    };
-  }
+  const filteredPlugins = plugins.filter((plugin) =>
+    Object.keys(queries).every((query: keyof PluginsByFilterType) => {
+      return typeof filters[query] === 'function' ? filters[query](plugin, queries[query]) : true;
```

Comment:
**Nitpick:** you can omit the `{ return ... }` here and just write the return value directly